src/features/web-compat.js

```diff
@@ -498,29 +477,55 @@ export class WebCompat extends ContentFeature {
     presentationFix () {
         try {
             // @ts-expect-error due to: Property 'presentation' does not exist on type 'Navigator'
-            if (window.navigator.presentation) {
+            if (window.navigator.presentation && import.meta.injectName !== 'integration') {
                 return
             }
 
-            this.defineProperty(window.navigator, 'presentation', {
-                value: {
-                },
-                writable: true,
-                configurable: true,
-                enumerable: true
+            const MyPresentation = class {
+                get defaultRequest () {
+                    return null
+                }
+
+                get receiver () {
+                    return null
+                }
+            }
+
+            // @ts-expect-error Presentation API is still experimental, TS types are missing
+            this.shimInterface('Presentation', MyPresentation, {
+                disallowConstructor: true,
+                allowConstructorCall: false,
+                wrapToString: true
             })
-            // @ts-expect-error due to: Property 'presentation' does not exist on type 'Navigator'
-            this.defineProperty(window.navigator.presentation, 'defaultRequest', {
-                value: null,
-                configurable: true,
-                enumerable: true
+
+            // @ts-expect-error Presentation API is still experimental, TS types are missing
+            this.shimInterface('PresentationAvailability', class {
+                // class definition is empty because there's no way to get an instance of it anyways
+            }, {
+                disallowConstructor: true,
+                allowConstructorCall: false,
+                wrapToString: true
             })
-            // @ts-expect-error due to: Property 'presentation' does not exist on type 'Navigator'
-            this.defineProperty(window.navigator.presentation, 'receiver', {
-                value: null,
-                configurable: true,
-                enumerable: true
+
+            // @ts-expect-error Presentation API is still experimental, TS types are missing
+            this.shimInterface('PresentationRequest', class {
+                // class definition is empty because there's no way to get an instance of it anyways
+            }, {
+                disallowConstructor: true,
+                allowConstructorCall: false,
+                wrapToString: true
             })
+
+            /** TODO: add shims for other classes in the Presentation API:
+             * PresentationConnection,
```

Comment:
Can you raise a task to track this?

Reply:
https://app.asana.com/0/1204556816597738/1207323488899175/f